common/persistence/serialization/serializer.go

```diff
@@ -120,17 +116,23 @@ type (
 	}
 
 	serializerImpl struct {
-		TaskSerializer
+		encodingType enumspb.EncodingType
 	}
 
 	marshaler interface {
 		Marshal() ([]byte, error)
 	}
 )
 
-// NewSerializer returns a PayloadSerializer
+// NewSerializer returns a Serializer that picks the encoding type based on an environment variable.
+// If none is set, it defaults to the Proto3 codec. See EncodingTypeFromEnv for details.
 func NewSerializer() Serializer {
-	return &serializerImpl{}
+	return NewSerializerWithEncoding(EncodingTypeFromEnv())
```

Comment:
The benefit from this approach that it'll always be the one from the env which I think is beneficial since there's no question of whether or not the env var takes effect.